File: 백준/Silver/1074. Z/Z.swift
import Foundation

let input = readLine()!.split(separator: " ").map{Int(String($0))!}
var N = input[0]
var r = input[1]
var c = input[2]

var res: Int = 0

func pos(max: Int, x: Int, y: Int) -> Int{
    
    if y < max / 2{
        return (x < max / 2) ? 1 : 3
    }
    else{
        return (x < max / 2) ? 2 : 4
    }
}

while N >= 1{
    let max = Int(pow(2.0, Double(N)))
    
    let num = pos(max: max, x: r, y: c)
    res += (num - 1) * (max / 2) * (max / 2)
    N -= 1
    if num == 2 || num == 4{
        c -= (max / 2)
    }
    if num == 3 || num == 4{
        r -= (max / 2)
    }
}

print(res)

File: 백준/Silver/1074. Z/Z.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int N = Integer.parseInt(st.nextToken());
		int y = Integer.parseInt(st.nextToken());
		int x = Integer.parseInt(st.nextToken());
		int startx = 0;
		int starty = 0;
		int cntx, cnty;
		int res = 0;
		
		while(N >= 1) {
			if(x >= startx && x < startx + Math.pow(2, N-1))
				cntx = 0;
			else
				cntx = 1;
			
			if(y >= starty && y < starty + Math.pow(2, N-1))
				cnty = 0;
			else
				cnty = 1;
			
			res += (cnty * 2 + cntx) * Math.pow(2, 2*N-2);
			
			startx += cntx * Math.pow(2, N-1);
			starty += cnty * Math.pow(2, N-1);
			N -= 1;
		}
		
		System.out.println(res);
	}
}